tools/ir-generator/irclass.cpp

```diff
@@ -133,6 +133,7 @@ void IrDefinitions::generate(std::ostream &t, std::ostream &out, std::ostream &i
         << std::endl
         << "namespace IR {\n"
         << "extern std::map<cstring, NodeFactoryFn> unpacker_table;\n"
+        << "using namespace P4::literals;\n"
```

Comment:
Let me double check

Reply:
So yes, it is in the header. We do have lots of cstring literals in `ir-generated.h`. Mostly around `toString` and friends. This could be changed to explicit `cstring::literal` call though as this is autogenerated code.

Reply:
Could be better to use `cstring::literal`, but I can live with the `using namespace` as it is in `IR` namespace.